content/to-other-languages/typescript-for-c-sharp-developers.md

```diff
@@ -4,3 +4,152 @@ showMetadata: false
 editable: true
 showToc: true
 ---
+
+# Overview
+
+TypeScript และ C# มาจากคนสร้างคนเดียวกัน คือ Microsoft ดังนั้น ลักษณะของ Syntax จึงมีความคล้ายคลึงสูงมาก แต่อย่างไรก็ตาม TypeScript ยังคงความเป็นเอกลักษณ์ของตัวภาษา Javascript อยู่
+
+ระบบ type ของ TypeScript ให้ประโยชน์หลายๆ อย่างเช่น ทำ code completion ดีขึ้น  สามารถหา error ได้ง่ายขึ้น และการสื่อสารระหว่างแต่ละส่วนของโปรแกรมค่อนข้างชัดเจน ว่า type อะไร
+
+**Note:**
+
+- ภาษา TypeScript สามารถทำงานร่วมกับ Javascript ได้เลย
+- ภาษา TypeScript มีความคล้ายคลึงกับ Javascript ES6 ขึ้นไป
+- การใช้ Library ของ TypeScript ร่วมกันกับ Javascript นั่นคือ `npm` ดังนั้น การเลือกใช้ Library ให้พึงสังเกตุว่า Library ตัวนั้นเขียนด้วย Typesctipt หรือไม่
+  - ถ้าเขียนด้วย Javascript โดยส่วนใหญ่จะเขียน Type  ของ Javascript แยกเป็นอีกไฟล์ อย่างไรก็ตาม การเรียกใช้ Library ลักษณะนี้อาจจะทำให้ Type ผิดพลาดได้ แต่มีโอกาสเจอน้อยนะ
+  - แต่ถ้าเขียนด้วย TypeScript เลย จะมีโอกาสเข้ากันได้ง่ายกว่า
+
+## Co-learning JavaScript
+
+If you’re a Java or C# programmer that is new to JavaScript in general, we recommend learning a little bit of JavaScript *without* types first to understand JavaScript’s runtime behaviors. Because TypeScript doesn’t change how your code *runs*, you’ll still have to learn how JavaScript works in order to write code that actually does something!
+
+ถ้าใครที่คุ้นเคยกับ JavaScript อยู่แล้ว ก็สามารถข้ามหัวพวกนี้ไปได้เลย
+
+### Asynchronous Programming
+
+**Read more**
+
+[Why Asynchronous? - NodeSource](https://nodesource.com/blog/why-asynchronous/)
+
+### Event Loop
+
+[Understanding the Node.js Event Loop - NodeSource](https://nodesource.com/blog/understanding-the-nodejs-event-loop/)
+
+[✨♻️ JavaScript Visualized: Event Loop - DEV Community](https://dev.to/lydiahallie/javascript-visualized-event-loop-3dif)
+
```

Comment:
Only one empty line should be okay because these three empty line will end up only one single empty line in HTML.